src/com/facebook/buck/jvm/kotlin/KotlincToJarStepFactory.java

```diff
@@ -82,21 +83,24 @@ public void createCompileStep(
       ImmutableList.Builder<Step> steps,
       BuildableContext buildableContext) {
 
-    steps.add(
-        new KotlincStep(
-            invokingRule,
-            outputDirectory,
-            sourceFilePaths,
-            pathToSrcsList,
-            ImmutableSortedSet.<Path>naturalOrder()
-                .addAll(
-                    Optional.ofNullable(extraClassPath.apply(buildContext))
-                        .orElse(ImmutableList.of()))
-                .addAll(declaredClasspathEntries)
-                .build(),
-            kotlinc,
-            extraArguments,
-            filesystem));
+    // Don't invoke kotlinc if we don't have any kotlin files.
+    if (sourceFilePaths.stream().anyMatch(KOTLIN_PATH_MATCHER::matches)) {
+      steps.add(
+          new KotlincStep(
+              invokingRule,
+              outputDirectory,
+              sourceFilePaths,
+              pathToSrcsList,
+              ImmutableSortedSet.<Path>naturalOrder()
+                  .addAll(
+                      Optional.ofNullable(extraClassPath.apply(buildContext))
+                          .orElse(ImmutableList.of()))
+                  .addAll(declaredClasspathEntries)
+                  .build(),
+              kotlinc,
+              extraArguments,
+              filesystem));
+    }
 
     ImmutableSortedSet<Path> javaSourceFiles =
```

Comment:
I thought about that as well but I realized that it actually needs the set to provide only the java source files to the java compiler. Kolinc actually needs both sets of source files so we didn't need a set of only kotlin files.